kernel/Build.mk

```diff
@@ -12,12 +12,31 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-.PHONY: kernel/build
-kernel/build: sandbox_setup
-	cd kernel && $(BWRAP) cargo build --release
+IMAGES=_a _b
 
-.PHONY: kernel/build-signed
-kernel/build-signed: kernel/build
+# ------------------------------------------------------------------------------
+# Macro to define targets for a specific board-app-and-image combination.
+# Arguments:
+# - $(IMAGE) [optional suffix to build A/B images]
+define KERNEL_IMAGE_TARGETS
+
+.PHONY: kernel/build$(IMAGE)
+kernel/build$(IMAGE): sandbox_setup
+	cd kernel && \
+		CARGO_TARGET_DIR="../build/kernel/cargo$(IMAGE)" \
+		RUSTFLAGS="-C link-arg=-T./layout$(IMAGE).ld" \
+		$(BWRAP) cargo build --release
+
+.PHONY: kernel/build-signed$(IMAGE)
+kernel/build-signed$(IMAGE): kernel/build$(IMAGE)
+
+endef
```

Comment:
Pernaps ` endef # KERNEL_IMAGE_TARGETS` would make this more readable?

Reply:
Good call. Done throughout.